### PR TITLE
dockerfile release

### LINKED ISSUE #5: [SPIKE] Containerized lambda
## Description
We will likely need to build a containerized lambda to deliver to APHL and need some research on how best to both develop and deploy.

## Considerations/Questions to think about
- We know the lambda function will need to read/write eICRs from an S3 bucket. How can we mock this for development? Eric mentioned `minIO`
- What are the tradeoffs of 1 Dockerfile vs 2 (one for the app and one that makes the app compatible with lambda
- Refiner is currently building images, pushing them to ghcr.io, and will use terraform on APHL's gitlab to pick up the image for deployment 
- Deployment for Refiner: https://github.com/georules/dibbs-aims-deployments/tree/main 
- How should we organize the repo to accomplish this?

## Task
Write up recommendations for how we should proceed, attaching it here and sharing in the eng channel. Ideally, we'd be able to write tickets based on the doc after it has been circulated among the engineers. The doc doesn't have to be tremendously long, just something we can use for documentation moving forward.

### PR DESCRIPTION
**Pull request by ericbuckley**
## Description
Adding a lambda Dockerfile and a process for building and releasing the application.

## Related Issues
closes #5 

## Additional Notes
[Add any additional context or notes that reviewers should know about.]

<--------------------- REMOVE THE LINES BELOW BEFORE MERGING --------------------->

## Checklist

Please review and complete the following checklist before submitting your pull request:

- [ ] I have ensured that the pull request is of a manageable size, allowing it to be reviewed within a single session.
- [ ] I have reviewed my changes to ensure they are clear, concise, and well-documented.
- [ ] I have updated the documentation, if applicable.
- [ ] I have added or updated test cases to cover my changes, if applicable.
- [ ] I have minimized the number of reviewers to include only those essential for the review.

## Checklist for Reviewers

Please review and complete the following checklist during the review process:

- [ ] The code follows best practices and conventions.
- [ ] The changes implement the desired functionality or fix the reported issue.
- [ ] The tests cover the new changes and pass successfully.
- [ ] Any potential edge cases or error scenarios have been considered.

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+      - name: Convert repository name to lowercase
+        run: echo "PACKAGE_NAME=$(echo '${{ github.repository }}' | tr '[:upper:]' '[:lower:]')" >> $GITHUB_ENV
+
+      - name: Build and push
+        uses: docker/build-push-action@v6
+        with:
+          platforms: linux/amd64
+          push: true
+          provenance: false
```
[the remark this answers is not in the document]
Comment (m-goggins) [reply .github/workflows/release.yml:53]: Huh that's an odd one and we might be able to figure out more from Frans and Geo if they have any insight.

[the remark this answers is not in the document]
Comment (m-goggins) [reply .github/workflows/release.yml:53]: Found `provenance ` guidance in the official docs, [here](https://docs.aws.amazon.com/lambda/latest/dg/python-image.html#python-image-instructions). See step 5 under "Creating an image from a base image": 

> Build the Docker image with the [docker build](https://docs.docker.com/engine/reference/commandline/build/) command. The following example names the image docker-image and gives it the test [tag](https://docs.docker.com/engine/reference/commandline/build/#tag). To make your image compatible with Lambda, you must use the --provenance=false option.
